src/mapping/Mapping.cpp

```diff
@@ -101,33 +102,48 @@ void Mapping::map(int inputDataID,
   PRECICE_ASSERT(output()->data(outputDataID)->values().size() / output()->data(outputDataID)->getDimensions() == static_cast<int>(output()->vertices().size()),
                  output()->data(outputDataID)->values().size(), output()->data(outputDataID)->getDimensions(), output()->vertices().size());
 
+  Sample sample{input()->data(inputDataID)->getDimensions(),
+                input()->data(inputDataID)->values(),
+                input()->data(inputDataID)->gradients()};
+  map(sample, output()->data(outputDataID)->values());
+}
+
+void Mapping::map(const Sample &input, Eigen::VectorXd &output)
+{
+  PRECICE_ASSERT(_hasComputedMapping);
+
   if (hasConstraint(CONSERVATIVE)) {
-    mapConservative(inputDataID, outputDataID);
+    mapConservative(input, output);
   } else if (hasConstraint(CONSISTENT)) {
-    mapConsistent(inputDataID, outputDataID);
+    mapConsistent(input, output);
   } else if (isScaledConsistent()) {
-    mapConsistent(inputDataID, outputDataID);
-    scaleConsistentMapping(inputDataID, outputDataID, getConstraint());
+    mapConsistent(input, output);
+    scaleConsistentMapping(input.values, output, getConstraint());
   } else {
     PRECICE_UNREACHABLE("Unknown mapping constraint.")
   }
 }
 
-void Mapping::scaleConsistentMapping(int inputDataID, int outputDataID, Mapping::Constraint constraint) const
+void Mapping::scaleConsistentMapping(const Eigen::VectorXd &input, Eigen::VectorXd &output, Mapping::Constraint constraint) const
 {
   PRECICE_ASSERT(isScaledConsistent());
+
+  if (input.size() == 0 || output.size() == 0) {
+    return;
+  }
```

Comment:
If only one of both is zero and the other one not, we should throw an assertion?

Reply:
True, this should always be true here.